doc/example_code/how_to_examples/controller.rst

```diff
@@ -0,0 +1,185 @@
++---------------+-------------------------------------------------------+
+| Back          | Called “select” or “share” on some controllers        |
++---------------+-------------------------------------------------------+
+| Guide         | Usually in the center, with a company logo            |
++---------------+-------------------------------------------------------+
+| LeftStick     | Pressing in on the left analog stick                  |
++---------------+-------------------------------------------------------+
+| RightStick    | Pressing in on the right analog stick                 |
++---------------+-------------------------------------------------------+
+| DPLeft        |                                                       |
++---------------+-------------------------------------------------------+
+| DPRight       |                                                       |
++---------------+-------------------------------------------------------+
+| DPUp          |                                                       |
++---------------+-------------------------------------------------------+
+| DPDown        |                                                       |
++---------------+-------------------------------------------------------+
+
+An event is dispatched when any of the values of the buttons are changed. These events can be handle like this:
+
+.. code-block:: python
+
+      @controller.event
+      def on_button_press(controller, button_name):
+          if button_name == 'a':
+              # start firing
+          elif button_name == 'b':
+              # do something else
+      @controller.event
+      def on_button_press(controller, button_name):
+          if button_name == 'a':
+              # start firing
+          elif button_name == 'b':
+              # do something else
+ 
+
+
+The (-1.0 to 1.0) Values On Controller
+======================================
+* The values will be between -1 and +1, with 0 being a centered joystick.
+* The x-axis numbers will be negative if the stick goes left, positive for right.
+* The y-axis numbers will be opposite of what you might expect. Negative for up, positive for down.
+
+.. list-table:: controller
+   :widths: 25 25 50
+   :header-rows: 1
+   
+   * - Left & Right
+     - Centered
+     - Up & Down
+   * -   (-1, 1)
+     -    0
+     -   (-1, 1)
+
+
+The controller Movements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. image:: controller.png
+    :width: 400px
+    :align: center
+    :alt: Screenshot of controller movements
+
+     
+Deadzone
+========
+A centered controller might have a value not at 0, but at 0.0001 or some small number. This will make for a small “drift” on a person’s character. We often counteract this by having a “dead zone” where if the number is below a certain value, we just assume it is zero to eliminate the drift.
+
+How we take care of the dead zone:
+
+After 
+
+.. code-block:: console
```

Comment:
Very small change, but this needs a blank line in between the code-block declaration and the contents, otherwise it raises a warning from Sphinx(and our CI treats warnings as errors).